From d7a726d0764197db64ca826bb16d92e7875682de Mon Sep 17 00:00:00 2001
From: Mart Raudsepp <leio@gentoo.org>
Date: Wed, 16 Jul 2014 16:25:16 +0300
Subject: [PATCH 1/3] gl: Check for raspberrypi-userland pkg-config files
 before manual checks

Some distributions or developers prefer to patch raspberrypi-userland to ship
with pkg-config files, or to add them manually. This uses them if present.

Note: Not a clean patch yet as its inclusion at this point is still doubtful
due to no raspberrypi-userland upstream inclusion of these pkg-config files
---
 configure.ac | 22 +++++++++++++++++++---
 1 file changed, 19 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 1cb7c88..4b73ebc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -698,8 +698,24 @@ case $host in
 
     dnl FIXME: EGL of RPi depends on GLESv1 or GLESv2
     dnl FIXME: GLESv2 of RPi depends on EGL... WTF!
-    LIBS="$LIBS -lvcos -lvchiq_arm"
-    AC_CHECK_LIB([bcm_host], [bcm_host_init],
+    PKG_CHECK_MODULES(BCM_HOST, bcm_host, [HAVE_BCM_HOST="yes"], [HAVE_BCM_HOST="no"])
+    AC_SUBST(BCM_HOST_LIBS)
+    AC_SUBST(BCM_HOST_CFLAGS)
+
+    PKG_CHECK_MODULES(EGL, egl, [HAVE_EGL="yes"], [HAVE_EGL="no"])
+    AC_SUBST(EGL_LIBS)
+    AC_SUBST(EGL_CFLAGS)
+
+    PKG_CHECK_MODULES(GLES2, glesv2, [HAVE_GLES2="yes"], [HAVE_GLES2="no"])
+    AC_SUBST(GLES2_LIBS)
+    AC_SUBST(GLES2_CFLAGS)
+
+    if test x"$HAVE_BCM_HOST" = x"yes"; then
+      HAVE_EGL_RPI=yes
+      AC_DEFINE(USE_EGL_RPI, [1], [Use RPi platform])
+    else
+     LIBS="$LIBS -lvcos -lvchiq_arm"
+     AC_CHECK_LIB([bcm_host], [bcm_host_init],
       [
         LIBS="$LIBS -lbcm_host"
         AC_CHECK_HEADER(bcm_host.h,
@@ -726,7 +742,7 @@ case $host in
               ])
           ])
       ])
-
+    fi
     LIBS=$old_LIBS
     CFLAGS=$old_CFLAGS
 
-- 
1.9.1

